libs/tup-components/src/mfa/MfaQRPanel.tsx

```diff
@@ -25,6 +35,21 @@ const MfaQRPanel: React.FC = () => {
           )}
         </div>
       </div>
+      {data && (
+        <div>
+          If your mobile device does not have a camera capable of scanning QR
+          codes, the following text code can be used:
+          <br />
+          <code>{data.otpkey.value_b32}</code> &nbsp;(
+          <Button
+            type="link"
+            onClick={() => navigator.clipboard.writeText(data.otpkey.value_b32)}
+          >
+            copy to clipboard
+          </Button>
+          )
```

Comment:
When testing locally, after clicking button to generate QR code, an error occurs, that complains that `data.optkey.value_b32` is undefined. The Portal UI disappears.


<details><summary>Error (click to reveal)</summary>

```
MfaQRPanel.tsx:43 Uncaught TypeError: Cannot read properties of undefined (reading 'value_b32')
    at MfaQRPanel (MfaQRPanel.tsx:43:30)
    at renderWithHooks (react-dom.development.js:16305:18)
    at updateFunctionComponent (react-dom.development.js:19588:20)
    at beginWork (react-dom.development.js:21601:16)
    at HTMLUnknownElement.callCallback2 (react-dom.development.js:4164:14)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:16)
    at invokeGuardedCallback (react-dom.development.js:4277:31)
    at beginWork$1 (react-dom.development.js:27451:7)
    at performUnitOfWork (react-dom.development.js:26557:12)
    at workLoopSync (react-dom.development.js:26466:5)
MfaQRPanel @ MfaQRPanel.tsx:43
renderWithHooks @ react-dom.development.js:16305
updateFunctionComponent @ react-dom.development.js:19588
beginWork @ react-dom.development.js:21601
callCallback2 @ react-dom.development.js:4164
invokeGuardedCallbackDev @ react-dom.development.js:4213
invokeGuardedCallback @ react-dom.development.js:4277
beginWork$1 @ react-dom.development.js:27451
performUnitOfWork @ react-dom.development.js:26557
workLoopSync @ react-dom.development.js:26466
renderRootSync @ react-dom.development.js:26434
performSyncWorkOnRoot @ react-dom.development.js:26085
flushSyncCallbacks @ react-dom.development.js:12042
(anonymous) @ react-dom.development.js:25651
setTimeout (async)
(anonymous) @ utils.ts:405
sleep @ utils.ts:404
scheduleMicrotask @ utils.ts:414
flush @ notifyManager.ts:60
batch @ notifyManager.ts:29
dispatch @ mutation.ts:334
execute @ mutation.ts:237
await in execute (async)
mutate @ mutationObserver.ts:135
(anonymous) @ useMutation.ts:109
onClick @ MfaQRPanel.tsx:27
onclick @ Button.tsx:71
callCallback2 @ react-dom.development.js:4164
invokeGuardedCallbackDev @ react-dom.development.js:4213
invokeGuardedCallback @ react-dom.development.js:4277
invokeGuardedCallbackAndCatchFirstError @ react-dom.development.js:4291
executeDispatch @ react-dom.development.js:9041
processDispatchQueueItemsInOrder @ react-dom.development.js:9073
processDispatchQueue @ react-dom.development.js:9086
dispatchEventsForPlugins @ react-dom.development.js:9097
(anonymous) @ react-dom.development.js:9288
batchedUpdates$1 @ react-dom.development.js:26140
batchedUpdates @ react-dom.development.js:3991
dispatchEventForPluginEventSystem @ react-dom.development.js:9287
dispatchEventWithEnableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay @ react-dom.development.js:6465
dispatchEvent @ react-dom.development.js:6457
dispatchDiscreteEvent @ react-dom.development.js:6430
MfaQRPanel.tsx:43 Uncaught TypeError: Cannot read properties of undefined (reading 'value_b32')
    at MfaQRPanel (MfaQRPanel.tsx:43:30)
    at renderWithHooks (react-dom.development.js:16305:18)
    at updateFunctionComponent (react-dom.development.js:19588:20)
    at beginWork (react-dom.development.js:21601:16)
    at HTMLUnknownElement.callCallback2 (react-dom.development.js:4164:14)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:16)
    at invokeGuardedCallback (react-dom.development.js:4277:31)
    at beginWork$1 (react-dom.development.js:27451:7)
    at performUnitOfWork (react-dom.development.js:26557:12)
    at workLoopSync (react-dom.development.js:26466:5)
MfaQRPanel @ MfaQRPanel.tsx:43
renderWithHooks @ react-dom.development.js:16305
updateFunctionComponent @ react-dom.development.js:19588
beginWork @ react-dom.development.js:21601
callCallback2 @ react-dom.development.js:4164
invokeGuardedCallbackDev @ react-dom.development.js:4213
invokeGuardedCallback @ react-dom.development.js:4277
beginWork$1 @ react-dom.development.js:27451
performUnitOfWork @ react-dom.development.js:26557
workLoopSync @ react-dom.development.js:26466
renderRootSync @ react-dom.development.js:26434
recoverFromConcurrentError @ react-dom.development.js:25850
performSyncWorkOnRoot @ react-dom.development.js:26096
flushSyncCallbacks @ react-dom.development.js:12042
(anonymous) @ react-dom.development.js:25651
setTimeout (async)
(anonymous) @ utils.ts:405
sleep @ utils.ts:404
scheduleMicrotask @ utils.ts:414
flush @ notifyManager.ts:60
batch @ notifyManager.ts:29
dispatch @ mutation.ts:334
execute @ mutation.ts:237
await in execute (async)
mutate @ mutationObserver.ts:135
(anonymous) @ useMutation.ts:109
onClick @ MfaQRPanel.tsx:27
onclick @ Button.tsx:71
callCallback2 @ react-dom.development.js:4164
invokeGuardedCallbackDev @ react-dom.development.js:4213
invokeGuardedCallback @ react-dom.development.js:4277
invokeGuardedCallbackAndCatchFirstError @ react-dom.development.js:4291
executeDispatch @ react-dom.development.js:9041
processDispatchQueueItemsInOrder @ react-dom.development.js:9073
processDispatchQueue @ react-dom.development.js:9086
dispatchEventsForPlugins @ react-dom.development.js:9097
(anonymous) @ react-dom.development.js:9288
batchedUpdates$1 @ react-dom.development.js:26140
batchedUpdates @ react-dom.development.js:3991
dispatchEventForPluginEventSystem @ react-dom.development.js:9287
dispatchEventWithEnableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay @ react-dom.development.js:6465
dispatchEvent @ react-dom.development.js:6457
dispatchDiscreteEvent @ react-dom.development.js:6430
react-dom.development.js:18687 The above error occurred in the <MfaQRPanel> component:

    at MfaQRPanel (http://localhost:3000/@fs/Users/wbomar/Code/tup-ui/libs/tup-components/src/mfa/MfaQRPanel.tsx:18:54)
    at div
    at MfaPairingLayout (http://localhost:3000/@fs/Users/wbomar/Code/tup-ui/libs/tup-components/src/mfa/MfaPairingView.tsx:15:29)
    at RenderedRoute (http://localhost:3000/@fs/Users/wbomar/Code/tup-ui/node_modules/.vite/tup-ui/deps/react-router-dom.js?v=45d12ca7:2538:5)
    at Outlet (http://localhost:3000/@fs/Users/wbomar/Code/tup-ui/node_modules/.vite/tup-ui/deps/react-router-dom.js?v=45d12ca7:2794:26)
    at MfaWrapper (http://localhost:3000/@fs/Users/wbomar/Code/tup-ui/libs/tup-components/src/mfa/MfaWrapper.tsx:15:23)
    at div
    at RequireAuth (http://localhost:3000/@fs/Users/wbomar/Code/tup-ui/libs/tup-components/src/utils/RequireAuth.tsx:13:24)
    at Mfa
    at RenderedRoute (http://localhost:3000/@fs/Users/wbomar/Code/tup-ui/node_modules/.vite/tup-ui/deps/react-router-dom.js?v=45d12ca7:2538:5)
    at Outlet (http://localhost:3000/@fs/Users/wbomar/Code/tup-ui/node_modules/.vite/tup-ui/deps/react-router-dom.js?v=45d12ca7:2794:26)
    at div
    at div
    at Layout (http://localhost:3000/@fs/Users/wbomar/Code/tup-ui/libs/tup-components/src/layout/PageLayout/PageLayout.tsx:12:19)
    at AppLayout
    at RenderedRoute (http://localhost:3000/@fs/Users/wbomar/Code/tup-ui/node_modules/.vite/tup-ui/deps/react-router-dom.js?v=45d12ca7:2538:5)
    at Routes (http://localhost:3000/@fs/Users/wbomar/Code/tup-ui/node_modules/.vite/tup-ui/deps/react-router-dom.js?v=45d12ca7:2854:5)
    at App
    at QueryClientProvider (http://localhost:3000/@fs/Users/wbomar/Code/tup-ui/node_modules/.vite/tup-ui/deps/@tanstack_react-query.js?v=45d12ca7:2770:3)
    at Router (http://localhost:3000/@fs/Users/wbomar/Code/tup-ui/node_modules/.vite/tup-ui/deps/react-router-dom.js?v=45d12ca7:2801:15)
    at BrowserRouter (http://localhost:3000/@fs/Users/wbomar/Code/tup-ui/node_modules/.vite/tup-ui/deps/react-router-dom.js?v=45d12ca7:3188:5)

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries.
overrideMethod @ console.js:213
logCapturedError @ react-dom.development.js:18687
update.callback @ react-dom.development.js:18720
callCallback @ react-dom.development.js:13923
commitUpdateQueue @ react-dom.development.js:13944
commitLayoutEffectOnFiber @ react-dom.development.js:23391
commitLayoutMountEffects_complete @ react-dom.development.js:24688
commitLayoutEffects_begin @ react-dom.development.js:24674
commitLayoutEffects @ react-dom.development.js:24612
commitRootImpl @ react-dom.development.js:26823
commitRoot @ react-dom.development.js:26682
performSyncWorkOnRoot @ react-dom.development.js:26117
flushSyncCallbacks @ react-dom.development.js:12042
(anonymous) @ react-dom.development.js:25651
setTimeout (async)
(anonymous) @ utils.ts:405
sleep @ utils.ts:404
scheduleMicrotask @ utils.ts:414
flush @ notifyManager.ts:60
batch @ notifyManager.ts:29
dispatch @ mutation.ts:334
execute @ mutation.ts:237
await in execute (async)
mutate @ mutationObserver.ts:135
(anonymous) @ useMutation.ts:109
onClick @ MfaQRPanel.tsx:27
onclick @ Button.tsx:71
callCallback2 @ react-dom.development.js:4164
invokeGuardedCallbackDev @ react-dom.development.js:4213
invokeGuardedCallback @ react-dom.development.js:4277
invokeGuardedCallbackAndCatchFirstError @ react-dom.development.js:4291
executeDispatch @ react-dom.development.js:9041
processDispatchQueueItemsInOrder @ react-dom.development.js:9073
processDispatchQueue @ react-dom.development.js:9086
dispatchEventsForPlugins @ react-dom.development.js:9097
(anonymous) @ react-dom.development.js:9288
batchedUpdates$1 @ react-dom.development.js:26140
batchedUpdates @ react-dom.development.js:3991
dispatchEventForPluginEventSystem @ react-dom.development.js:9287
dispatchEventWithEnableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay @ react-dom.development.js:6465
dispatchEvent @ react-dom.development.js:6457
dispatchDiscreteEvent @ react-dom.development.js:6430
Show 1 more frame
react-dom.development.js:12056 Uncaught TypeError: Cannot read properties of undefined (reading 'value_b32')
    at MfaQRPanel (MfaQRPanel.tsx:43:30)
    at renderWithHooks (react-dom.development.js:16305:18)
    at updateFunctionComponent (react-dom.development.js:19588:20)
    at beginWork (react-dom.development.js:21601:16)
    at beginWork$1 (react-dom.development.js:27426:14)
    at performUnitOfWork (react-dom.development.js:26557:12)
    at workLoopSync (react-dom.development.js:26466:5)
    at renderRootSync (react-dom.development.js:26434:7)
    at recoverFromConcurrentError (react-dom.development.js:25850:20)
    at performSyncWorkOnRoot (react-dom.development.js:26096:20)
```

</details>

https://github.com/TACC/tup-ui/assets/62723358/05c1c7d8-f48d-4e70-a56d-8eeb50fbee2e

Reply:
P.S. I found to test my changes, so far, despite this obstruction.

Reply:
I have a solution (in #270 0b50542): a new different message for user if `data.optkey` is unavailable.